public/app/features/dashboard/services/DashboardLoaderSrv.ts

```diff
@@ -8,51 +8,57 @@ import { backendSrv } from 'app/core/services/backend_srv';
 import impressionSrv from 'app/core/services/impression_srv';
 import kbn from 'app/core/utils/kbn';
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import { getGrafanaStorage } from 'app/features/storage/storage';
+import { DashboardDTO, DashboardRoutes } from 'app/types';
 
 import { appEvents } from '../../../core/core';
 
 import { getDashboardSrv } from './DashboardSrv';
 
 export class DashboardLoaderSrv {
   constructor() {}
-  _dashboardLoadFailed(title: string, snapshot?: boolean) {
-    snapshot = snapshot || false;
+  _dashboardLoadFailed(title: string, snapshot?: boolean): DashboardDTO {
     return {
       meta: {
         canStar: false,
-        isSnapshot: snapshot,
+        isSnapshot: snapshot || false,
         canDelete: false,
         canSave: false,
         canEdit: false,
         dashboardNotFound: true,
       },
       dashboard: { title },
-    };
+    } as any;
```

Comment:
The existing types (or really lack of types!) for this make it weird.  `dashboardNotFound` is a flag that has special behavior 🤷